### PR TITLE
importinto: cleanup sorted files, writer memory quota, and test

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #46704

Problem Summary:

- cleanup sorted files when task done
- calculate writer memory quota based on index count
- test

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Review by okJiang — COMMENTED** [review]
rest lgtm

File: disttask/importinto/dispatcher.go [hunk 1]
```diff
@@ -641,6 +686,28 @@ func (dsp *ImportDispatcherExt) failJob(ctx context.Context, taskHandle dispatch
 	)
 }
 
+func cleanUpGlobalSortedData(ctx context.Context, gTask *proto.Task, taskMeta *TaskMeta) {
+	// we can only clean up files after all write&ingest subtasks are finished,
+	// since they might share the same file.
+	// we don't return error here, since the task is already done, we should
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply disttask/importinto/dispatcher.go:692]: yes

File: disttask/importinto/encode_and_sort_operator.go [hunk 2]
```diff
@@ -25,15 +25,24 @@ import (
...
 	"go.uber.org/zap"
 )
 
 const (
 	maxWaitDuration = 30 * time.Second
+
+	// We limit the memory usage of KV deliver to 1GB per concurrency, and data
+	// KV deliver has external.DefaultMemSizeLimit, the rest of memory is for
+	// all index KV deliver.
+	// Note: this size the memory taken by KV, not the size of taken by golang,
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply disttask/importinto/encode_and_sort_operator.go]: https://github.com/pingcap/tidb/pull/47092/commits/7999a5173d33113f9b18d80465f53a9b1a655edc

File: disttask/importinto/dispatcher.go [hunk 3]
```diff
@@ -561,9 +571,38 @@ func updateResult(handle dispatcher.TaskHandle, gTask *proto.Task, taskMeta *Tas
...
+		if err = json.Unmarshal(bs, &subtaskMeta); err != nil {
+			return 0, err
+		}
+		subtaskMetas = append(subtaskMetas, &subtaskMeta)
+	}
+	var loadedRowCount uint64
+	for _, subtaskMeta := range subtaskMetas {
+		loadedRowCount += subtaskMeta.Result.LoadedRowCnt
+	}
+	return loadedRowCount, nil
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply disttask/importinto/dispatcher.go]: https://github.com/pingcap/tidb/pull/47092/commits/7999a5173d33113f9b18d80465f53a9b1a655edc

**Review by okJiang — APPROVED** [review]

**Pull request merged**